website/src/components/canvas/sidebar/ResizeTextfields.jsx

```diff
@@ -46,11 +46,50 @@ class ResizeTextfields extends React.Component {
 
     constructor(props) {
         super(props);
+        this.state = {
+          disableTextfields: true,
+         };
         this.generateButtonHandler = this.generateButtonHandler.bind(this);
         this.handlerTextfieldHeight = this.handlerTextfieldHeight.bind(this);
         this.handlerTextfieldWidth = this.handlerTextfieldWidth.bind(this);
     }
-
+    /**
+     * Function to automatically be performed once the component mount.
+     * Used here to initialize canvas elements if not done and if the new canvas id
+     * is not null
+     * @returns {void}
+     */
+    componentDidMount() {
```

Comment:
Instead of making disableTextfields part of state I think it would be cleaner if you just made a constant for it in render.
i.e.
```
const disableTextfields = this.props.targetedId !== null;
```
Then use this constant. That way you don't have to add all of these functions and rely on the component having state.